locust_plugins/listeners.py

```diff
@@ -130,22 +133,34 @@ def on_start(self, environment: locust.env.Environment):
             sys.exit(1)
         self.set_gitrepo()
 
-        if is_worker() or is_master():
+        if is_worker(self.env) or is_master(self.env) or type(environment.runner) == LocalRunner:
             # swarm generates the run id for its master and workers
             if getattr(environment.parsed_options, "run_id", False):
                 self._run_id = parser.parse(environment.parsed_options.run_id)
+            elif hasattr(self, "_run_id"):
+                pass
             else:
                 logging.info(
                     "You are running distributed, but without swarm. run_id:s in Timescale will not match exactly between load gens"
                 )
                 self._run_id = datetime.now(timezone.utc)
+                logging.info(f"Run id is {self._run_id}")
         else:
             # non-swarm runs need to generate the run id here
             self._run_id = datetime.now(timezone.utc)
-        if not is_worker():
+            logging.info(f"Run id is {self._run_id}")
+        if not is_worker(self.env):
             logging.info(
                 f"Follow test run here: {self.env.parsed_options.grafana_url}&var-testplan={self._testplan}&from={int(self._run_id.timestamp()*1000)}&to=now"
             )
+            if type(environment.runner) != LocalRunner:
```

Comment:
I think LocalRunner can send messages to itself, so a lot of this logic may not be necessary. Try it out at least.

Reply:
i had tried it, but get no client for localrunner. so added the condition.
Traceback (most recent call last):
  File "/home/recolada/Desktop/locust/.venv/lib/python3.10/site-packages/locust/event.py", line 40, in fire
    handler(**kwargs)
  File "/home/recolada/Desktop/locust/.venv/lib/python3.10/site-packages/locust_plugins/listeners.py", line 159, in on_start
    for _, worker in enumerate(environment.runner.clients):
AttributeError: 'LocalRunner' object has no attribute 'clients'

Reply:
(no need for a for-loop or anything)

Reply:
yes!! i changed it as per comment.